### PR TITLE
feat(core): create new button-group component

### PR DESCRIPTION
**Pull request by emmoldovan**

### CODE REVIEW

File: core/src/components/cat-button-group/cat-button-group.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  render() {
+    return (
+      <Host role="group" aria-label={this.a11yLabel}>
+        <slot></slot>
+      </Host>
+    );
+  }
+
+  componentDidLoad() {
+    this.formElements = Array.from(this.hostElement.querySelectorAll('cat-button')) as HTMLCatButtonElement[];
```
Comment (fynnfeldpausch) [inline core/src/components/cat-button-group/cat-button-group.tsx:35]: I think the typecast at the end can be deleted

File: core/src/components/cat-button-group/cat-button-group.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    this.formElements.forEach((element, index) => {
+      element.buttonGroupPosition = index === 0 ? 'first' : index === this.formElements.length - 1 ? 'last' : 'middle';
```
Comment (fynnfeldpausch) [inline core/src/components/cat-button-group/cat-button-group.tsx:38]: not sure if this is a problem but this will only be executed once at the beginning. This means if we have a button group in angular with an ngIf on some buttons, the position will be wrong if the dynamically show up. Do you think we should execute this on ever componentDidRender?

File: core/src/components/cat-button/cat-button.scss [hunk 3]
```diff
@@ -78,6 +78,29 @@ $button-sizes: (
...
+  }
+
+  &-middle {
+    border-radius: 0;
+  }
+
+  &-last {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+    border-right: none;
```
Comment (fynnfeldpausch) [inline core/src/components/cat-button/cat-button.scss:96]: im confused. why does the last button have no right border?

**Review by fynnfeldpausch — CHANGES_REQUESTED** [review]

File: core/src/components/cat-button-group/cat-button-group.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  display: inline-flex;
+  vertical-align: middle;
+}
+
+::slotted(cat-button[variant='outlined']),
+::slotted(cat-button:not([variant])) {
+  margin-right: -1px;
+}
+
+::slotted(cat-button[variant='outlined']:first-child),
```
Comment (fynnfeldpausch) [inline core/src/components/cat-button-group/cat-button-group.scss:12]: why is this needed?

File: core/src/components/cat-button-group/cat-button-group.tsx (hunk 2 above)
Comment (fynnfeldpausch) [reply core/src/components/cat-button-group/cat-button-group.tsx:31]: wow, didnt know this is an option.

File: core/src/components/cat-button/cat-button.scss [hunk 5]
```diff
@@ -78,6 +78,32 @@ $button-sizes: (
...
+  &-middle {
+    border-radius: 0;
+  }
+
+  &-last {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+
+  &:focus-visible {
```
Comment (fynnfeldpausch) [inline core/src/components/cat-button/cat-button.scss:98]: shouldn't this be after the &:hover rule? otherwise if there is focus and we hover, it will be back to 1

**Review by fynnfeldpausch — CHANGES_REQUESTED** [review]

File: core/src/components/cat-button-group/cat-button-group.scss (hunk 4 above)
Comment (emmoldovan) [reply core/src/components/cat-button-group/cat-button-group.scss:12]: Yeap, leftovers

**Review by fynnfeldpausch — APPROVED** [review]
